--- origsrc/lxpanel-0.3.8.1/configure.ac	2008-07-07 00:37:14.000000000 -0500
+++ src/lxpanel-0.3.8.1/configure.ac	2009-02-15 14:25:37.562500000 -0600
@@ -93,7 +93,7 @@ fi
 
 LXPANEL_MODULE="-avoid-version \
 -rpath \$(libdir)/lxpanel/plugins \
--no-undefined "'-export-symbols-regex "^[[^_]].*"'
+-no-undefined -Wl,\$(top_builddir)/src/liblxpanel.a "'-export-symbols-regex "^[[^_]].*"'
 AC_SUBST(LXPANEL_MODULE)
 
 AC_ARG_ENABLE([alsa],
--- origsrc/lxpanel-0.3.8.1/src/Makefile.am	2008-04-20 03:44:13.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/Makefile.am	2009-02-15 14:30:43.609375000 -0600
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = plugins
+SUBDIRS = . plugins
 
 bin_PROGRAMS = lxpanel lxpanelctl
 
@@ -44,7 +44,7 @@ lxpanel_SOURCES = \
 	gtk-run.c
 
 if BUILD_PLUGIN_LOADER
-DYNAMIC_FLAGS = -export-dynamic
+DYNAMIC_FLAGS = -Wl,--export-all-symbols -Wl,--out-implib,liblxpanel.a
 endif
 lxpanel_LDFLAGS = $(DYNAMIC_FLAGS)
 
@@ -58,3 +58,6 @@ lxpanelctl_LDADD = -lX11 $(INTLLIBS)
 
 EXTRA_DIST = \
 	plugins/icon.xpm
+
+$(top_builddir)/src/plugins/libbuiltin_plugins.a:
+	cd $(@D); $(MAKE) $(@F)
