kbled.c: ERR(x) macro defined in "dbg.h"

--- origsrc/lxpanel-0.3.8.1/src/plugins/cpu/cpu.c	2008-04-20 03:44:13.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/plugins/cpu/cpu.c	2009-02-15 15:19:36.453125000 -0600
@@ -24,7 +24,9 @@
 #include <string.h>
 #include <sys/time.h>
 #include <time.h>
+#ifndef __CYGWIN__
 #include <sys/sysinfo.h>
+#endif
 #include <stdlib.h>
 #include <glib/gi18n.h>
 
--- origsrc/lxpanel-0.3.8.1/src/plugins/kbled/kbled.c	2008-07-02 21:28:36.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/plugins/kbled/kbled.c	2009-02-15 15:25:54.921875000 -0600
@@ -32,6 +32,7 @@
 #include "panel.h"
 #include "misc.h"
 #include "plugin.h"
+#include "dbg.h"
 /*
 enum {
     CapsLock = 0,
--- origsrc/lxpanel-0.3.8.1/src/plugins/volume/volume-impl.c	2008-04-20 03:44:13.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/plugins/volume/volume-impl.c	2009-02-15 14:25:37.578125000 -0600
@@ -29,7 +29,7 @@
 
 #include "volume-impl.h"
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__CYGWIN__)
 #include <sys/soundcard.h>
 #elif defined(__linux__) || defined(__Linux__)
 #include <linux/soundcard.h>
