--- origsrc/lxpanel-0.3.8.1/src/configurator.c	2008-06-10 22:51:19.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/configurator.c	2009-02-15 14:59:30.484375000 -0600
@@ -46,6 +46,7 @@ void panel_configure(Panel* p, int sel_p
 void restart(void);
 void gtk_run(void);
 void panel_config_save(Panel* panel);
+void save_global_config(void);
 static void logout(void);
 
 Command commands[] = {
--- origsrc/lxpanel-0.3.8.1/src/misc.h	2008-07-02 18:27:34.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/misc.h	2009-02-15 14:48:17.203125000 -0600
@@ -80,6 +80,8 @@ void calculate_position(Panel *np);
 gchar *expand_tilda(gchar *file);
 GdkPixbuf *gdk_pixbuf_scale_ratio(GdkPixbuf *p, int width, int height, GdkInterpType itype,
                                   gboolean keep_ratio);
+void _gtk_image_set_from_file_scaled(GtkWidget* img, const gchar *file, gint width,
+                                     gint height, gboolean keep_ratio);
 GtkWidget *_gtk_image_new_from_file_scaled(const gchar *file, gint width, gint height,
                                           gboolean keep_ratio);
 GtkWidget *_gtk_image_new_from_file_scaled(const gchar *file, gint width,
--- origsrc/lxpanel-0.3.8.1/src/panel.h	2008-06-10 22:51:19.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/panel.h	2009-02-15 15:11:44.531250000 -0600
@@ -186,6 +186,7 @@ extern FbEv *fbev;
 
 #define FBPANEL_WIN(win)  gdk_window_lookup(win)
 
+void panel_destroy(Panel *p);
 void panel_set_wm_strut(Panel *p);
 void panel_set_dock_type(Panel *p);
 void panel_set_orientation(Panel *p);
--- origsrc/lxpanel-0.3.8.1/src/systray/fixedtip.c	2008-04-20 03:44:13.000000000 -0500
+++ src/lxpanel-0.3.8.1/src/systray/fixedtip.c	2009-02-15 14:50:27.546875000 -0600
@@ -26,6 +26,8 @@ static GtkWidget *label = NULL;
 static int screen_width = 0;
 static int screen_height = 0;
 
+void fixed_tip_hide (void);
+
 static gboolean
 button_press_handler (GtkWidget *tip,
                       GdkEvent  *event,
