--- origsrc/lxpanel-0.5.4.1/src/plugins/volume/volume-impl.c	2009-04-20 10:07:31.000000000 -0500
+++ src/lxpanel-0.5.4.1/src/plugins/volume/volume-impl.c	2010-02-10 01:02:33.655380800 -0600
@@ -30,7 +30,7 @@
 
 #include "volume-impl.h"
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__CYGWIN__)
 #include <sys/soundcard.h>
 #elif defined(__linux__) || defined(__Linux__)
 #include <linux/soundcard.h>
